### PR TITLE
Add optional initial temperature guess

### LINKED ISSUE #32: Add option to pass `T_initial_guess` to `PhaseEquil`
This will allow us to use the previous temperature as an initial guess in saturation adjustment.

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR:
 - Adds an optional initial temperature guess
 - Adds appropriate docs
 - Fixes some docs

@tapios, I've added a test that I think may be useful for us to review interactively.

Closes #32.

### CODE REVIEW

File: test/relations.jl [hunk 1]
```diff
@@ -1614,5 +1617,49 @@ end
...
+    ts =
+        PhaseEquil_ρeq.(
+            param_set,
+            ρ,
+            e_int,
+            q_tot,
+            8,
+            FT(1e-1),
+            RS.NewtonsMethod,
+            T .+ (FT(0.1) .- rand(FT, length(T))),
```
Comment (tapios) [inline test/relations.jl:1635]: D you want this to be unbiased w.r.t. to the true solution, i.e.,
```suggestion
            T .+ (FT(0.1) .* rand(FT, length(T))),
```

File: test/relations.jl [hunk 2]
```diff
@@ -1614,5 +1617,49 @@ end
...
+    ts =
+        PhaseEquil_ρeq.(
+            param_set,
+            ρ,
+            e_int,
+            q_tot,
+            2,
+            FT(1e-1),
+            RS.NewtonsMethod,
+            T .+ (FT(0.5) .- rand(FT, length(T))), # provide an initial guess +/- 0.5 K from true solution
```
Comment (tapios) [inline test/relations.jl:1656]: As above, how about this:
```suggestion
            T .+ (FT(0.5) .* rand(FT, length(T))), # provide an initial guess +/- 0.5 K from true solution
```

**Review by tapios — APPROVED** [review]
LGTM. Some comments on tests below. 

Would it be difficult to set maxiter by default to 1 is T_guess if given, and to something larger otherwise? If so, we may get away with a reasonable O(0.1 K) convergence tolerance in all cases.

File: test/relations.jl (hunk 1 above)
Comment (charleskawczynski) [reply test/relations.jl:1635]: Ah, yes. This was a last minute typo, thanks!

**Pull request merged**